llvm/tools/llvm-link/llvm-link.cpp

```diff
@@ -393,8 +393,17 @@ static bool linkFiles(const char *argv0, LLVMContext &Context, Linker &L,
   // Similar to some flags, internalization doesn't apply to the first file.
   bool InternalizeLinkedSymbols = false;
   for (const auto &File : Files) {
+    auto ErrOrExpected = MemoryBuffer::getFileOrSTDIN(File);
+
+    // When we encounter a missing file, print all missing files to stderr.
+    if (auto EC = ErrOrExpected.getError())
+      if (EC == std::errc::no_such_file_or_directory)
+        for (auto &F : Files)
+          if (!llvm::sys::fs::exists(F))
+            errs() << "No such file or directory: '" << F << "'\n";
+
     std::unique_ptr<MemoryBuffer> Buffer =
-        ExitOnErr(errorOrToExpected(MemoryBuffer::getFileOrSTDIN(File)));
+        ExitOnErr(errorOrToExpected(std::move(ErrOrExpected)));
```

Comment:
Unsure if it's necessary to error on all missing files at the same time. I would recommend just returning an error on this file if it's found, and using something like `createStringError("No such file or directory: %s", F.str())`.

Reply:
> Unsure if it's necessary to error on all missing files at the same time.

I have no strong preference here, as long as I get the filename instead of nothing.
**Q: So you'd prefer only one error message mentioning the first missing file?**

> I would recommend just returning an error on this file if it's not found [...]

Like this? (If not: what do you mean by 'returning'?)
```
  // When we encounter a missing file, print all missing files to stderr.
  if (auto EC = ErrOrExpected.getError())
    if (EC == std::errc::no_such_file_or_directory)
      ExitOnErr(createStringError(EC, "No such file or directory: %s", File.c_str()));
```

Reply:
Great, thank you!